src/DAQProcess.cc

```diff
@@ -30,18 +31,26 @@ DAQProcess::DAQProcess(std::list<std::string> args) {
 
 void DAQProcess::register_modules(std::unique_ptr<ModuleList> const& ml) { ml->ConstructGraph(bufferMap_, userModuleMap_, commandOrderMap_); }
 
-void DAQProcess::execute_command(std::string cmd) {
+void DAQProcess::execute_command(const std::string & cmd) {
     std::unordered_set<std::string> user_module_list;
     for (auto const& um : userModuleMap_) {
         user_module_list.insert(um.first);
     }
 
+    std::vector< std::pair<std::string, std::future<std::string> > > cmd_results ;
+
     TLOG(TLVL_DEBUG) << "Executing Command " << cmd << " for UserModules defined in the CommandOrderMap";
     if (commandOrderMap_.count(cmd)) {
         for (auto& moduleName : commandOrderMap_[cmd]) {
             if (userModuleMap_.count(moduleName)) {
-                userModuleMap_[moduleName]->execute_command(cmd);
-                user_module_list.erase(moduleName);
+	        try { 
+	            cmd_results.push_back( std::make_pair( moduleName, userModuleMap_[moduleName]->execute_command(cmd) ) ) ; 
+	        }
+	        catch ( ... ) {
+		  TLOG(TLVL_WARNING) << moduleName << " throw an exception while receiving command " << cmd ;
```

Comment:
This is a thorny problem, but what is any external entity supposed to do with this info?

Presumably this log message makes it's way to CCM (eventually) and CCM reacts or ignores it.

For this to have a chance of being usefully processed I think at least the following info needs to be added

* an identifier for the exception type that was caught
* some identifier of the DAQ process as a whole, both its type (program name) and it's unique ID.

Reply:
Hi Brett, yes. At the moment we don't have specific suggestions from the CCM. This is our expected behaviour. 

I agree with both your suggestions. Specifically.
> an identifier for the exception type that was caught

This is in the pipeline. Alessandro has the exception based code mentioned by Giovanna and once we are having a look at it, we will start implementing the dedicated specifications. Until then I'll prefer to keep this item on hold.

>some identifier of the DAQ process as a whole, both its type (program name) and it's unique ID. 

This is in fact very much desirable as it is one of the requirements from CCM we discussed at the CERN workshop. I'll try to see if I can code something simply that can be easily added to the DAQProcess interface.